packages/nimble-components/src/rich-text-editor/testing/rich-text-editor.pageobject.ts

```diff
@@ -0,0 +1,66 @@
+import type { RichTextEditor } from '..';
```

Comment:
We have developed some practices for pageobjects that should be followed here such as not exposing element references directly, creating helpers for different actions, and encapsulating dom manipulation. The goal is for page objects to de-couple tests from the DOM implementation details.

We should also be looking to follow the table patterns where the page object is exported from nimble-components and nimble-angular so that they can be used from tests in SLE.

@jattasNI and @m-akinc will help give context of how to move more into the page object

Reply:
Milan covered the basics but I'll add some details. The goal is to use the page object to abstract away implementation details and instead expose an API that will be stable even if the DOM structure changes. This means we should not expose `Element` types on the API and instead expose methods that interact with those elements (e.g. `clickBoldButton()` or return primitive types with specific information about those elements (e.g. `getBoldButtonName(): string`.

Some other details those methods should abstract away include:
1. timing details like the need to call `await waitForUpdatesAsync();`.
2. event details like the need to call `dispatchEvent()`
3. specific class names and selectors like `bullet-list`

The end result should be tests that are very easy to read because they are calling well-named high-level methods on the page object instead of doing the implementation-specific things above.

Reply:
The [table pageobject](https://github.com/ni/nimble/blob/main/packages/nimble-components/src/table/testing/table.pageobject.ts) is large but can be used as an example for the patterns discussed.

Reply:
Thanks for the details and references, I have completely refactored the page objects to not return the `Element` types, instead exposing necessary methods to interact with them. Please take a look at those methods and test cases and let me know if we should consider updating them.

Reply:
It's looking a lot cleaner, thanks! One more suggestion (assuming it's still necessary to have methods that interact with all the buttons after addressing my other feedback): rather than using class names to identify buttons, the page object could export an enum with one item for each button. That way it could encapsulate the class names to prevent tests from needing to update if they change.

Reply:
Thanks for the suggestion. I have added a const value that exports the button index with respect to its positioning in the toolbar which will identify each button instead of using class names and do the necessary interactions through the methods.

Reply:
+1 for enums. just to be clear you will need to use nimble's enum style by making a `testing/types.ts` and using the format described in the docs: https://github.com/ni/nimble/blob/main/packages/nimble-components/docs/coding-conventions.md#use-const-objects-instead-of-typescript-enums

Reply:
Thanks for the reference. I have added a `types.ts` file for the toolbar button along with the spec file.

Reply:
@vivinkrishna-ni reopening, please notice carefully how both of our descriptions stated the method needs to take an enum as the type. Your methods are still taking a number as the type. Please address this in a follow-up pr

Reply:
Apologies for the mistake. I have updated it in the next PR of the rich text editor. https://github.com/ni/nimble/pull/1424